src/main/kotlin/org/rust/ide/annotator/RustErrorAnnotator.kt

```diff
@@ -106,18 +103,26 @@ class RustErrorAnnotator : Annotator {
         val notImplemented = mustImplement.keys - implemented.keys
         if (!notImplemented.isEmpty()) {
             val toImplement = trait.functionList.filter { it.name in notImplemented }
+            val implHeaderTextRange = TextRange.create(
+                impl.textRange.startOffset,
+                impl.type?.textRange?.endOffset ?: impl.textRange.endOffset
+            )
 
             holder.createErrorAnnotation(implHeaderTextRange,
-                "Not all trait items implemented, missing: `${notImplemented.first()}`"
+                "Not all trait items implemented, missing: ${notImplemented.namesList} [E0046]"
             ).registerFix(ImplementMethodsFix(impl, toImplement))
-
         }
 
         val notMembers = implemented.filterKeys { it !in canImplement }
         for (method in notMembers.values) {
             holder.createErrorAnnotation(method.identifier,
-                "Method is not a member of trait `${trait.name}`")
+                "Method `${method.name}` is not a member of trait `$traitName` [E0407]")
         }
+
+        implemented
+            .map { it.value to canImplement[it.key] }
```

Comment:
Yeah, you can't destruct in lambdas at the moment (this will be fixed in Kotlin 1.1 I believe), so it has to be more ugly:

```
.mapNotNull {
  val (name, fn) = it
  canImplement[fn]?.let { fn to it }  
}
```